base/src/main/java/com/smartdevicelink/managers/screen/choiceset/DeleteChoicesOperation.java

```diff
@@ -93,23 +103,57 @@ public void onFinished() {
                     public void onResponse(int correlationId, RPCResponse response) {
                         if (!response.getSuccess()) {
                             if (completionListener != null) {
-                                completionListener.onComplete(false);
+                                completionListener.onComplete(false, loadedCells);
                             }
                             DebugTool.logError(TAG, "Failed to delete choice: " + response.getInfo() + " | Corr ID: " + correlationId);
 
                             DeleteChoicesOperation.super.onFinished();
+                        } else {
+                            if (loadedCells != null) {
+                                loadedCells.remove(loadedCellFromChoiceId(correlationId));
```

Comment:
I found an issue, Steps to show error:
Step 1: Upload a `ChoiceSet`.
Step 2: Present `ChoiceSet` and select an item.
Step 3: In your `ChoiceSetSelectionListener` retrieve the list of uploaded `ChoiceCell`s and delete one of them.
Step 4: Present `ChoiceSet`

Observed: 
Errors: 

```
2021-09-13 15:58:07.635 28442-28442/com.sdl.hellosdlandroid E/PreloadPresentChoicesOperation: 5.2.0: Presenting Choice set failed: null
2021-09-13 15:58:07.635 28442-28442/com.sdl.hellosdlandroid E/BaseChoiceSetManager: 5.2.0: Present finished but an unhandled state occurred and callback failed
```

Expected: 
The deleted ChoiceCell should be uploaded again and the ChoiceSet should be present.

I traced it back to here, the Correlation ID was 53 and the ChoiceCell ID was 3.

Code I used for testing:
Add this code to `onChoiceSelected` in hello_sdl

```
 HashSet<ChoiceCell> choiceCellHashSet = sdlManager.getScreenManager().getPreloadedChoices();
 List<ChoiceCell> choiceCellList = new ArrayList<>(choiceCellHashSet);
 sdlManager.getScreenManager().deleteChoices(Collections.singletonList(choiceCellList.get(1)));
```